### ISSUE TITLE
TypeConverters not call with string[]

### ISSUE THREAD

**Issue by potnoddle** [JoshClose/CsvHelper#1838]
Tring to convert a string to an string array, using a TypeConverter

Create a type converter and a POC class, see below. Reading a csv file I would expect to see the string converted to an array, of what ever object I want. Instead the converter isn't called and the field is null. Other TypeConverters work for instance I've create a NullableIntConverter and a NullableBoolConverter.

I'm guessing that certain fields are filtered out before parsing, maybe a check if they contain a TypeConvert attribute first.

```
    public class StringArrayConverter : ITypeConverter
    {
        public object ConvertFromString(string text, IReaderRow row, MemberMapData memberMapData)
        {
            if (!string.IsNullOrEmpty(text))
            {
                return text.Split('|');
            }
            else
            {
                return new string[] { };
            }
        }

        public string ConvertToString(object value, IWriterRow row, MemberMapData memberMapData)
        {
            var list = value as string[];
            if (value != null)
            {
                return string.Join('|', list);
            }
            else
            {
                return string.Empty;
            }
        }
    }
```

```
    public class StringArrayPOC
    {
        public string Code { get; set; }
        [TypeConverter(typeof(StringArrayConverter))]
        public string[] StringArray { get; set; }
    }
```

```
Code,StringArray 
a,"a|b"
ab,"ab|bb|cc|dd"
C,
```

**Comment by JoshClose** [comment]
You have something else going on. Your setup should work fine.

```cs
void Main()
{
    var s = new StringBuilder();
    s.Append("Code,StringArray\r\n");
    s.Append("a,\"a|b\"\r\n");
    s.Append("ab,\"ab|bb|cc|dd\"\r\n");
    s.Append("C,\r\n");
    var config = new CsvConfiguration(CultureInfo.InvariantCulture)
    {
    };
    using (var reader = new StringReader(s.ToString()))
    using (var csv = new CsvReader(reader, config))
    {
        csv.GetRecords<StringArrayPOC>().ToList().Dump();
    }
}

public class StringArrayPOC
{
    public string Code { get; set; }
    [TypeConverter(typeof(StringArrayConverter))]
    public string[] StringArray { get; set; }
}

public class StringArrayConverter : ITypeConverter
{
    public object ConvertFromString(string text, IReaderRow row, MemberMapData memberMapData)
    {
        if (!string.IsNullOrEmpty(text))
        {
            return text.Split('|');
        }
        else
        {
            return new string[] { };
        }
    }

    public string ConvertToString(object value, IWriterRow row, MemberMapData memberMapData)
    {
        var list = value as string[];
        if (value != null)
        {
            return string.Join('|', list);
        }
        else
        {
            return string.Empty;
        }
    }
}
```

![image](https://user-images.githubusercontent.com/114971/126798400-50685122-2980-4175-ab70-ae31c120205f.png)

**Comment by potnoddle** [comment]
Thanks for the reply, how did you host your implementation? Mine was in a console app, originally targeting dot.net 6, however I extracted it into a separate project that can target different environment. I tried without success on 4.8, 3.1 core I even created a win forms version, see attached zip.

[CsvHelperTest.zip](https://github.com/JoshClose/CsvHelper/files/6872239/CsvHelperTest.zip)

Additionally, I did notice that you have ArrayConverter and IEnumerableConverter, implementations. Which I also tried without luck.

Finally I've tried throw an debugging/exception inside my converter and I don't see any evidence it even access it.

Is there a way for returning debugging info, as I'm at a loss as how to proceeded.
